crates/taplo-lsp/src/handlers/formatting.rs

```diff
@@ -43,6 +50,15 @@ pub(crate) async fn format<E: Environment>(
     ws.taplo_config
         .update_format_options(&doc_path, &mut format_opts);
 
+    let scopes = ws.taplo_config.format_scopes(&doc_path).collect::<Vec<_>>();
+    tracing::trace!(
+        ?doc_path,
+        ?format_opts,
+        ?scopes,
+        all_rules = ?ws.taplo_config.rule,
+        matched_rules = ?ws.taplo_config.rules_for(&doc_path).collect::<Vec<_>>(),
+    );
```

Comment:
I'm assuming this is for debugging. Unless you have a strong opinion about it, I would suggest keeping `ws.taplo_config.format_scopes(&doc_path)` inlined below (as it was before) and not round-trip through a `Vec` for the sake of tracing at trace level.

Reply:
Hey @ia0 thanks for the review! I do remember the scopes feeling somewhat important to preserve so they show up in the LSP output tab with e.g. a vscode config setting like:

```json
  "evenBetterToml.taplo.environment": {
    "RUST_LOG": "trace"
  }
```

But, you'll have to forgive me if I'm not able to recall the precise problem I thought I was solving. 

Either way, I hear you about wanting to avoid the round trip unless we're going to be tracing: what do you think of 6031848 ?

Reply:
Thanks! This looks good to me.